content/blog/bare-metal-provisioning-on-hpe-greenlake-using-terraform.md

```diff
@@ -0,0 +1,359 @@
+   ```
+2. #### Configure the Terraform provider.
+
+   Create an empty folder and put a file in it called main.tf with the following contents:
+
+   **main.tf:**
+
+   ```hcl
+   terraform {
+     required_providers {
+       hpegl = {
+         source  = "HPE/hpegl"
+         version = ">= 0.3.12"
+       }
+     }
+   }
+
+   provider "hpegl" {    
+     # metal block for configuring bare metal resources.
+     metal {   
+     } 
+   }
+   ```
+
+   T﻿his tells Terraform that you are going to be using HPE/hpegl as your provider and you are using bare metal service.
+
+## Write resource configuration for Compute Instance creation
+
+To deploy compute instance, you need to use the **hpegl_metal_host** terraform resource.
+
+***Note**:*
+
+* *compute instance is AKA host.* 
+* *compute instance type is AKA machine size.*
+
+The **hpegl_metal_host** resource supports many different arguments, but these are the required ones:
+
+* networks – List of networks. The list must always include any networks marked as '**required**' for that location. 
+* machine_size – Compute Instance type.
+* ssh – List of SSH keys to push to the host.
+* location – The data center location of where the compute instance will be provisioned.
+* Image – OS image in the form flavor@version.
+
+  Y﻿ou can also check the documentation [here](https://registry.terraform.io/providers/HPE/hpegl/latest/docs/resources/metal_host) to see all the required and optional fields.
+
+  Your next step with the TF file is to query the HPE GreenLake provider to collect the above-required information for creating a host. For this, you will use the Terraform data statements.  
+
+### Querying for available OS images
+
+In order to list the available OS images for OS, add the below data statements in your terraform file ***main.tf***:
+
+```hcl
+data "hpegl_metal_available_images" "ubuntu" { 
+  # select anything that looks like ubuntu:20.04
+  filter {
+    name   = "flavor"
+    values = ["(?i)ubuntu"] 
+  }
+
+  filter {
+    name   = "version"
+    values = ["20.04*"] 
+  }
+}
+```
+
+ The OS image list can be fetched using the following statements in main.tf: 
+
+```hcl
+locals {
+  ubuntu_image = format("%s@%s", data.hpegl_metal_available_images.ubuntu.images[0].flavor,
+                          data.hpegl_metal_available_images.ubuntu.images[0].version)
+}
+```
+
+### Querying for other available resources
+
+For this, you should use **hpegl_metal_available_resources** data resource. For example, the following statements show how to retrieve and store the available SSH keys in a local variable.\
+\
+Append to main.tf:
+
+```hcl
+# query available resources.
+data "hpegl_metal_available_resources" "available" {
+}
+
+# using one of the available SSH keys.
+locals  {
+  ssh_keys = data.hpegl_metal_available_resources.available.ssh_keys[0].name
+}
+```
+
+Using a similar technique, you can retrieve the rest of the data you need - networks, machine size, etc.\
```

Comment:
Hmmm... I see it here, too, so maybe this is intentional...
![image](https://user-images.githubusercontent.com/77496261/222976199-44f2fb7d-8918-4e62-a14d-6fd578f2ae24.png)